appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -377,12 +377,119 @@ class CreateUserOnBuddyRequestTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(richard)
 	}
 
+	def 'Overwrite user created on buddy request'()
+	{
+		given:
+		def richard = addRichard()
+		def mobileNumberBob = "+$timestamp"
+		def inviteURL = buildInviteUrl(sendBuddyRequestForBob(richard, mobileNumberBob))
+
+		when:
+		def response = appService.requestOverwriteUser(mobileNumberBob)
+
+		then:
+		response.status == 200
+
+		User bob = appService.addUser(this.&assertUserOverwriteResponseDetails, "B o b", "Bob Changed",
+				"Dunn Changed", "BD Changed", mobileNumberBob, ["overwriteUserConfirmationCode": "1234"])
+		bob
+		bob.firstName == "Bob Changed"
+		bob.lastName == "Dunn Changed"
+		bob.nickname == "BD Changed"
+		bob.mobileNumber == mobileNumberBob
+		bob.goals.size() == 1 //mandatory goal
+		bob.goals[0].activityCategoryUrl == GAMBLING_ACT_CAT_URL
+
+		def buddiesRichard = appService.getBuddies(richard)
+		buddiesRichard.size() == 0
+
+		def buddiesBob = appService.getBuddies(bob)
+		buddiesBob.size() == 0
+
+		def getMessagesResponse = appService.getMessages(richard)
+		getMessagesResponse.status == 200
+		getMessagesResponse.responseData._embedded
+		getMessagesResponse.responseData._embedded."yona:messages".size() == 1
+
+		def buddyConnectResponseMessages = getMessagesResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "BuddyConnectResponseMessage"}
+		def buddyConnectResponseMessage = buddyConnectResponseMessages[0]
+		buddyConnectResponseMessage.message == "User account was deleted"
+		buddyConnectResponseMessage.nickname == "Bob Dunn"
+		buddyConnectResponseMessage._links.self.href.startsWith(richard.messagesUrl)
+		buddyConnectResponseMessage._links?."yona:process"?.href?.startsWith(getMessagesResponse.responseData._embedded."yona:messages".findAll{ it."@type" == "BuddyConnectResponseMessage"}[0]._links.self.href)
+
+		def getResponse = appService.getMessages(richard)
```

Comment:
This is a bit duplicate. The `buddyConnectResponseMessage` and `disconnectMessage` are equal, isn't it?